### PR TITLE
Add warning about unsupported tools

### PR DESCRIPTION
**Pull request by ianjevans**
Clarify that support for unsupported tools, drivers, and frameworks is "best effort."

Fixes DOC-7613.

- [ ] Backport to currently supported versions after content review.

### CODE REVIEW

**Comment by dikshant** [comment]
We will wanna wait a bit on this, we decided to make some changes to the language, I'll update with the new copy as soon as I have it.

File: src/current/_includes/v23.1/misc/tooling.md
Comment (rimadeodhar) [inline src/current/_includes/v23.1/misc/tooling.md:15]:
```suggestion
Cockroach Labs provides 'best effort' support for tools, drivers, and frameworks that are not officially supported. This means that while we will do our best to assist you, we may not be able to fully troubleshoot errors in your deployment.
```

File: src/current/_includes/v23.1/misc/tooling.md
Comment (rimadeodhar) [inline src/current/_includes/v23.1/misc/tooling.md:16]: @dikshant: would it also be worth adding some warning that around moving production workloads to CockroachDB using unsupported drivers?

**Comment by rimadeodhar** [comment]
Thanks @ianjevans for working on this and apologies for the delay. I'll let @dikshant take a look as well.

File: src/current/_includes/v23.1/misc/tooling.md
Comment (dikshant) [reply src/current/_includes/v23.1/misc/tooling.md:16]: Yeah, for paying customers we should suggest they evaluate this with their account team before considering to do this.

Comment (dikshant) [reply src/current/_includes/v23.1/misc/tooling.md:16]: We should also add a bit about raising unsupported tooling related issues with the tool maintainers directly.

**Review by dikshant — APPROVED** [review]
LGTM aside from a few comments I left above.

**Pull request merged**
